vertical-pod-autoscaler/pkg/recommender/input/cluster_feeder.go

```diff
@@ -431,7 +436,7 @@ func (feeder *clusterStateFeeder) getSelector(vpa *vpa_types.VerticalPodAutoscal
 			{conditionType: vpa_types.ConfigDeprecated, delete: true},
 		}
 	}
-	if legacySelector != nil {
+	if legacySelector != nil && *beta1APIDeprecated {
```

Comment:
This code can be easily tested in cluster_feeder_test.go

Reply:
Especially when the tests reveal you have a bug :no_mouth:. Added.